src/bg_daemon/fetchers/imgurfetcher.py

```diff
@@ -226,27 +227,69 @@ def _select_image(self, galleries):
                 except IndexError:
                     return None
 
+            # if the "image" is actually an album, get the topmost image from
+            # the album
+            if isinstance(selected_image, GalleryAlbum):
+                selected_image = self._get_image_from_album(selected_image)
+                if selected_image is None:
+                    continue
+
+            logger.debug("Selecting Image {}".format(selected_image.title))
             attempts += 1
             if attempts > 30:
                 return None
 
             if selected_image.width < self.min_width:
+                logger.debug("Rejecting due to width...")
                 continue
 
             if selected_image.height < self.min_height:
+                logger.debug("Rejecting due to height...")
                 continue
 
-            for word in self.blacklist_words:
-                if word in selected_image.title:
-                    continue
+            bad_title = False
+            if self.blacklist_words is not None:
+                for word in self.blacklist_words:
+                    if word in selected_image.title:
+                        bad_title = True
+                        break
 
-                if word in selected_image.title:
-                    continue
+                    if selected_image.description is not None:
+                        if word in selected_image.description:
+                            bad_title = True
+                            break
 
-                elected = True
+            if bad_title:
+                logger.debug("Rejecting due to blacklist_words...")
+                continue
+
+            elected = True
+
+        logger.debug("selected image {}".format(selected_image))
 
         return selected_image
 
+    """
+        _get_image_from_album
+
+        If the query results in an album, get an image from it.
+    """
+    def _get_image_from_album(self, album):
+
+        assert(isinstance(album, GalleryAlbum))
```

Comment:
I don't have a strong preference. Maybe start with generic exceptions
(ValueError) and descriptive messages or if you can identify something
common to all these errors, you should probably create an Exception
class to encompass them.

On (Apr-08-15| 6:24), Santiago Torres wrote:
> >  
> >          return selected_image
> >  
> > +    """
> > +        _get_image_from_album
> > +
> > +        If the query results in an album, get an image from it.
> > +    """
> > +    def _get_image_from_album(self, album):
> > +
> > +        assert(isinstance(album, GalleryAlbum))
> 
> To do this, do you suggest we define our own exceptions or raise a general Exception with a descriptive message?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/SantiagoTorres/bg_daemon/pull/4/files#r27968616